### PR TITLE
Make the server options configuration into individual settings

### LINKED ISSUE #20: Split `options` setting into individual settings
The `options` hash is hard to document and can confuse users. Further, it exposes some internals that may not be easy to understand.

The [code](https://github.com/logstash-plugins/logstash-output-email/blob/master/lib/logstash/outputs/email.rb#L115-L122) references certain options specifically, so we should just expose this stuff directly in the plugin with individual settings

Related: https://github.com/elastic/logstash/issues/2267
Moved from: https://github.com/elastic/logstash/issues/3242

### PR DESCRIPTION
**Pull request by purbon**
This PR introduced a normalization for the server configuration options by making them individual options in the plugin, this provides more clarity to the way this options should be used by the end user.

Fixes #20 and #24

### CODE REVIEW

**Comment by purbon** [comment]
@jordansissel your thoughts and review here would be much appreciate it.

**Comment by jordansissel** [comment]
Will review this soon (tomorrow likely). <3

**Comment by purbon** [comment]
Thanks @jordansissel much appreciate it!

**Comment by purbon** [comment]
@guyboertje your thoughts also super welcome!

File: lib/logstash/outputs/email.rb [hunk 1]
```diff
@@ -131,7 +107,7 @@ def register
         delivery_method :@via, options
```
Comment (guyboertje) [inline lib/logstash/outputs/email.rb]: I know its not part of your refactor, but does this line actually work? I would have thought it should be `@via.to_sym`.

Comment (purbon) [inline lib/logstash/outputs/email.rb]: not sure this works either, could you open an issue for this and I will make sure to track this down in a next PR? Thanks  @guyboertje

**Comment by guyboertje** [comment]
LGTM - existing users need to alter their config, though - should this be released as v 1.1.0?

**Comment by purbon** [comment]
@guyboertje I would release this together with #13 as of 2.0 branch, I understand this changes also backwards compatibility as people should update their way to handle configuration. what do you think?

Thanks for your feedback man!

**Comment by guyboertje** [comment]
@purbon - v 1.1.0 is fine.

**Pull request merged**